### PR TITLE
Make `KotlinDslScriptsModelSpec` a crossVersionTest

### PR DESCRIPTION
**Pull request by reinsch82**
Addresses 
- #27198

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test ACT

File: platforms/core-configuration/kotlin-dsl-tooling-builders/build.gradle.kts [hunk 1]
```diff
@@ -5,19 +5,19 @@ plugins {
...
-    implementation(project(":model-core"))
-    implementation(project(":core"))
     implementation(project(":resources"))
     implementation(project(":platform-base"))
     implementation(project(":platform-jvm"))
-    implementation(project(":plugins-java"))
     implementation(project(":plugins-java-base"))
     implementation(project(":tooling-api"))
     implementation(project(":logging"))
+    implementation(project(":kotlin-dsl-tooling-models"))
```
Comment (donat) [inline platforms/core-configuration/kotlin-dsl-tooling-builders/build.gradle.kts:20]: ❌  Why do you need to introduce dependencies for the main source set when your change only affects tests?

**Review by donat — CHANGES_REQUESTED** [review]
I have concerns with the newly introduced dependencies, otherwise the PR is good to go.

**Review by donat — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle merge

**Pull request merged**